x-pack/plugins/secops/public/components/page/hosts/events_table/index.tsx

```diff
@@ -80,37 +144,43 @@ const getEventsColumns = (startDate: number, endDate: number) => [
     sortable: true,
     truncateText: true,
     hideForMobile: true,
-    render: (item: EventItem) => (
-      <ProviderContainer>{getOr('--', 'event.type', item)}</ProviderContainer>
-    ),
+    render: ({ event }: { event: Ecs }) => <>{getOr('--', 'event.type', event)}</>,
   },
   {
     name: 'Source',
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'source.ip', item).slice(0, 12)} : {getOr('--', 'source.port', item)}
+        {formatSafely('source.ip', event)} : {getOr('--', 'source.port', event)}
       </>
     ),
   },
   {
     name: 'Destination',
     sortable: true,
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'destination.ip', item).slice(0, 12)} : {getOr('--', 'destination.port', item)}
+        {formatSafely('destination.ip', event)} : {getOr('--', 'destination.port', event)}
       </>
     ),
   },
   {
     name: 'Location',
     sortable: true,
     truncateText: true,
-    render: (item: EventItem) => (
+    render: ({ event }: { event: Ecs }) => (
       <>
-        {getOr('--', 'geo.region_name', item)} - {getOr('--', 'geo.country_iso_code', item)}
+        {getOr('--', 'geo.region_name', event)} : {getOr('--', 'geo.country_iso_code', event)}
       </>
     ),
   },
 ];
+
+const formatSafely = (path: string, data: Ecs) => {
+  if (has('source.ip', event)) {
```

Comment:
Did you mean to have `path` sent here in has as in `has(path`?

Right now it looks like it is showing a bunch of `--` when it should be showing their destination ip's. This looks like the root of the issue. Probably unit tests around formatSafely? :-)

<img width="1309" alt="screen shot 2019-01-09 at 3 46 37 pm" src="https://user-images.githubusercontent.com/1151048/50934653-7b369c00-1426-11e9-92b1-35d093d9aff6.png">

Reply:
push a fix with unit testing